plugins/trendmicro_visionone/icon_trendmicro_visionone/connection/connection.py

```diff
@@ -0,0 +1,54 @@
+import insightconnect_plugin_runtime
+from insightconnect_plugin_runtime.exceptions import (
+    ConnectionTestException,
+    PluginException,
+)
+from .schema import ConnectionSchema, Input
+
+# Custom imports below
+import pytmv1
+
+
+class Connection(insightconnect_plugin_runtime.Connection):
+    def __init__(self):
+        super(self.__class__, self).__init__(input=ConnectionSchema())
+
+    def connect(self, params: dict = None):
+        """
+        Connection config params are supplied as a dict in
+        params or also accessible in self.parameters['key']
+
+        The following will setup the var to be accessed
+          self.blah = self.parameters['blah']
+        in the action and trigger files as:
+          blah = self.connection.blah
+        """
+        self.logger.info("Connect: Connecting...")
+
+        url = params.get(Input.API_URL, "https://tmv1-mock.trendmicro.com")
+        key = params.get(Input.API_KEY).get("secretKey", "Dummy-API-KEY")
```

Comment:
Is it a good idea to have a default of `Dummy-API-KEY` here?

Reply:
Indeed it was not the best idea. Thank you for pointing it out.
I have replaced it with another, perhaps better solution, but do let me know if you have a specific expectation there that you would like to see there implemented instead. :)